numpy/core/tests/test_numeric.py

```diff
@@ -1965,57 +1965,146 @@ def test_base_range(self):
             np.base_repr(1, 37)
 
 
+# we pre-create arrays as we sometime want to pass the same instance
+# and sometime not. Passing the same instances may not mean the array are
+# equal, especially when containing None
+
+
+# those are 0-d arrays, it used to be a special case
+# where (_e0 == _e0).all() would raise
+_e0 = np.array(0, dtype='int')
+_e1 = np.array(1, dtype='float')
+
+# Non-nanable – those cannot hold nans
+_a12 = np.array([1, 2])
+_a12b = _a12.copy()
+_a123 = np.array([1, 2, 3])
+_a13 = np.array([1, 3])
+_a34 = np.array([3, 4])
+
+
+# those can hold nan
+_aS1 = np.array(["a"], dtype="S1")
+_aS1b = _aS1.copy()
+_aS1u4 = np.array([("a", 1)], dtype="S1,u4")
+_aS1u4b = _aS1u4.copy()
+
+
+_b1 = np.array([1, 2, np.nan])
+_b2 = np.array([1, np.nan, 2])
+_b3 = np.array([1, 2, np.inf])
+_b4 = np.array(np.nan)
+
+_t1 = _b1.astype("timedelta64")
+_t2 = _b2.astype("timedelta64")
+
+_md1 = np.array([[0, 1], [np.nan, 1]])
+
+# I belive this is a mistake, as both cplx ends up being nan+nan.j (same
+# complex in the list) twice instead of haveig nan+1.j, 1+nan.j
+_cplx1, _cplx2 = [np.array([1 + 1j])] * 2
+_cplx1.real, _cplx2.imag = np.nan, np.nan
+
+# this is I believe what was originally meant
+_cplx3, _cplx4 = np.complex64(1, np.nan), np.complex64(np.nan, 1)
```

Comment:
I think @rossbar wrote the original test, maybe you remember what you meant ?

Reply:
Ack yes, good catch. The intent was to assert that arrays are equal when either the real *or* the imaginary parts are nan, so the original test should be something like:

```python
>>> a = np.array([1 + 1j])
>>> b = a.copy()
>>> a.real = np.nan
>>> a
array([nan+1.j])
>>> b.image = np.nan
>>> b
array([1.+nanj])
>>> np.array_equal(a, b, equal_nan=True)
True
```

The [existing test](https://github.com/numpy/numpy/blame/5ca26b1b9df5e8c89923a2bb723a5ab95b6fe09b/numpy/core/tests/test_numeric.py#L2014-L2018) is correct, but indeed is not as specific as intended!

Reply:
Great ! I'll update the message !